### PR TITLE
Implementing climb limit switch

### PR DESCRIPTION
**Pull request by VZamora5**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/ClimbSubsystem.java [hunk 1]
```diff
@@ -43,6 +47,8 @@ public ClimbSubsystem() {
     leadClimbRotate = new WPI_TalonSRX(LEAD_ROTATE_MOTOR);
     followClimbRotate = new WPI_TalonSRX(FOLLOW_ROTATE_MOTOR);
 
+    climbLimitSwitch = new DigitalInput(3);
```
Comment (hazeltorek) [inline src/main/java/frc/robot/subsystems/ClimbSubsystem.java]: Use a constant!! !! !!!!!

**Review by hazeltorek — CHANGES_REQUESTED** [review]
test it babey!! ! !!! wooo yeah!!!! scringle

**Comment by jboutin020106** [comment]
Looks good to me.

**Review by jboutin020106 — APPROVED** [review]
Everything seems correct

File: src/main/java/frc/robot/subsystems/ClimbSubsystem.java [hunk 2]
```diff
@@ -43,8 +47,14 @@ public ClimbSubsystem() {
     leadClimbRotate = new WPI_TalonSRX(LEAD_ROTATE_MOTOR);
     followClimbRotate = new WPI_TalonSRX(FOLLOW_ROTATE_MOTOR);
 
+    climbLimitSwitch = new AnalogInput(3);
```
Comment (hazeltorek) [inline src/main/java/frc/robot/subsystems/ClimbSubsystem.java]: the constant was added but you didn't change this to use the constant

File: src/main/java/frc/robot/subsystems/ClimbSubsystem.java [hunk 3]
```diff
@@ -168,6 +182,10 @@ public void setClimbMode(boolean mode) {
     climbMode = mode;
   }
 
+  public boolean getLimitState(){
+    return (climbLimitSwitch.getValue() > 50);
+  }
+
```
Comment (hazeltorek) [inline src/main/java/frc/robot/subsystems/ClimbSubsystem.java]: write a javadoc comment explaining why this is a comparison

File: src/main/java/frc/robot/subsystems/ClimbSubsystem.java [hunk 4]
```diff
@@ -189,9 +208,8 @@ public void initSendable(SendableBuilder sendable) {
     sendable.addBooleanProperty("Climb mode", this::getClimbMode, this::setClimbMode);
     sendable.addDoubleProperty("Left lift encoder", leftClimbLiftEncoder::getIntegratedSensorPosition, null);
     sendable.addDoubleProperty("Right lift encoder", rightClimbLiftEncoder::getIntegratedSensorPosition, null);
-    sendable.addDoubleProperty("Rotate encoder", () -> {
-      return encoderTicksToDegrees(leadClimbRotate.getSelectedSensorPosition());
-    }, null);
+    sendable.addDoubleProperty("Rotate encoder", () -> encoderTicksToDegrees(leadClimbRotate.getSelectedSensorPosition()), null);
+    sendable.addDoubleProperty("Climb limit switch", climbLimitSwitch::getValue, null);
```
Comment (hazeltorek) [inline src/main/java/frc/robot/subsystems/ClimbSubsystem.java]: change this to a boolean property that references `getLimitState()`

**Review by hazeltorek — CHANGES_REQUESTED** [review]

**Review by hazeltorek — APPROVED** [review]
waiting on test to merge

**Comment by hazeltorek** [comment]
it works but we need to redo the angles maybe?

File: src/main/java/frc/robot/subsystems/ClimbSubsystem.java [hunk 5]
```diff
@@ -26,9 +29,16 @@ public class ClimbSubsystem extends SubsystemBase {
...
 
+  private SensorCollection pulse;
+
   private WPI_TalonSRX leadClimbRotate;
   private WPI_TalonSRX followClimbRotate;
 
+  private AnalogInput climbLimitSwitch;
+
+  private double position;
+  double convertedPos;
```
Comment (hazeltorek) [inline src/main/java/frc/robot/subsystems/ClimbSubsystem.java]: private

File: src/main/java/frc/robot/Constants.java [hunk 6]
```diff
@@ -133,12 +133,15 @@ public static final class ClimbConstants {
         public static final double LIFT_MIN_POSITION = 0;
         public static final double LIFT_MAX_POSITION = 225559;
 
-        public static final double ROTATE_MIN_ANGLE = -4.75;
-        public static final double ROTATE_MAX_ANGLE = 23.0;
+        public static final double ROTATE_MIN_ANGLE = 3556;
+        public static final double ROTATE_MAX_ANGLE = 4100;
```
Comment (hazeltorek) [inline src/main/java/frc/robot/Constants.java]: comment with units here

File: src/main/java/frc/robot/subsystems/ClimbSubsystem.java [hunk 7]
```diff
@@ -43,8 +53,16 @@ public ClimbSubsystem() {
     leadClimbRotate = new WPI_TalonSRX(LEAD_ROTATE_MOTOR);
     followClimbRotate = new WPI_TalonSRX(FOLLOW_ROTATE_MOTOR);
 
+    pulse = new SensorCollection(leadClimbRotate);
```
Comment (hazeltorek) [inline src/main/java/frc/robot/subsystems/ClimbSubsystem.java]: is this the preferred way to instantiate these? i think this should be the following

```java
pulse = leadClimbRotate.getSensorCollection();
```

not sure if thats the correct method but same idea

File: src/main/java/frc/robot/subsystems/ClimbSubsystem.java [hunk 8]
```diff
@@ -189,9 +223,10 @@ public void initSendable(SendableBuilder sendable) {
     sendable.addBooleanProperty("Climb mode", this::getClimbMode, this::setClimbMode);
     sendable.addDoubleProperty("Left lift encoder", leftClimbLiftEncoder::getIntegratedSensorPosition, null);
     sendable.addDoubleProperty("Right lift encoder", rightClimbLiftEncoder::getIntegratedSensorPosition, null);
-    sendable.addDoubleProperty("Rotate encoder", () -> {
-      return encoderTicksToDegrees(leadClimbRotate.getSelectedSensorPosition());
-    }, null);
+    sendable.addDoubleProperty("Raw rotate encoder", leadClimbRotate::getSelectedSensorPosition, null);
+    sendable.addDoubleProperty("Rotate encoder", () -> encoderTicksToDegrees(leadClimbRotate.getSelectedSensorPosition()), null);
+    sendable.addDoubleProperty("Position Var", () -> position, null);
```
Comment (hazeltorek) [inline src/main/java/frc/robot/subsystems/ClimbSubsystem.java]: when completed values from testing should be removed and any values that are staying should be adjusted to make sure they have correct units

**Review by hazeltorek — CHANGES_REQUESTED** [review]

**Review by hazeltorek — APPROVED** [review]

**Review by hazeltorek — APPROVED** [review]

**Pull request merged**